### PR TITLE
[WIP] Implement swing/twist decomposition

### PR DESCRIPTION
**Pull request by richard-uk1**
This PR implements a decomposition of a quaternion into a swing and a twist.

I have realised that I don't actually need this functionality to do what I want, but since I have implemented it I thought I would submit a PR to see if there was interest in merging it. If there is, and people have valid use cases, I can finish off handling the degenerate case.

### CODE REVIEW

File: src/quaternion.rs [hunk 1]
```diff
@@ -173,6 +169,40 @@ impl<S: BaseFloat> Quaternion<S> {
...
+    /// let swing = rotation * twist.conjugate();
+    /// // if we recombine our rotations we get the original back.
+    /// assert_ulps_eq!(rotation, swing * twist);
+    /// panic!("swing: {:#?}, twist: {:#?}, rotation: {:#?}, new rotation: {:#?}",
+    /// Euler::from(swing), Euler::from(twist), Euler::from(rotation), Euler::from(swing * twist));
+    /// # }
+    /// ```
+    /// . There is a degenerate case where the direction is orthogonal to the direction of the
+    /// quaternion, and the angle is 180 degrees. In this case there are infinite possible
+    /// decompositions, so allow the user to specify one with `fallback`.
```
Comment (brendanzab) [inline src/quaternion.rs]: I like this documentation! ❤️  - Wondering if it would be better just to return an `Option` in that case?

Comment (derekdreery) [reply src/quaternion.rs]: Yeah could easily do that, if it's preferable. What do you think about the utility of the whole method?

Comment (derekdreery) [reply src/quaternion.rs]: I think I'd like to leave this open until someone who understands this stuff has a look and says whether it's useful or not.

**Comment by derekdreery** [comment]
I'm closing this since peeps seem to be coalescing around nalgebra

**Pull request closed without merging**